plutus-core/plutus-ir/test/transform/inline/letTypeApp2

```diff
@@ -1,21 +1,16 @@
 {- 
 funApp gets inlined because it only occurs once.
-k should not get inlined by the unconditional inline optimization because it occurs more
-than once.
-Both instances of k should get inlined by the optimization that inlines fully applied functions 
-because it is fully applied.
-The first instance of k being inlined verifies that an over-applied function still gets inline with
-type applications.
+Both instances of k get inlined by unconditional inliner.
```

Comment:
The original purpose of this test is to test callsite inlining. If everything gets inlined unconditionally then it defeats the purpose.

Reply:
Yes the purpose has changed, but I left this here because it's another example of type applications.